yb-voyager/src/namereg/namereg.go

```diff
@@ -0,0 +1,527 @@
+	return fmt.Sprintf("multiple matching %s names: %s", e.ObjectType, strings.Join(e.Names, ", "))
+}
+
+type ErrNameNotFound struct {
+	ObjectType string
+	Name       string
+}
+
+func (e *ErrNameNotFound) Error() string {
+	return fmt.Sprintf("%s name not found: %s", e.ObjectType, e.Name)
+}
+
+func matchName(objType string, names []string, name string) (string, error) {
+	if name[0] == '"' {
+		if name[len(name)-1] != '"' {
+			return "", fmt.Errorf("invalid quoted %s name: [%s]", objType, name)
+		}
+		name = name[1 : len(name)-1]
+	}
+	var candidateNames []string
+	for _, n := range names {
+		if n == name { // Exact match.
+			return n, nil
+		}
+		if strings.EqualFold(n, name) {
+			candidateNames = append(candidateNames, n)
+		}
+	}
+	if len(candidateNames) == 1 {
+		return candidateNames[0], nil
+	}
+	if len(candidateNames) > 1 {
+		return "", &ErrMultipleMatchingNames{ObjectType: objType, Names: candidateNames}
+	}
+	return "", &ErrNameNotFound{ObjectType: objType, Name: name}
+}
+
+//================================================
+
+type identifier struct {
+	Quoted, Unquoted, MinQuoted string
+}
+
+// Can be a name of a table, sequence, materialised view, etc.
+type ObjectName struct {
+	SchemaName        string
+	FromDefaultSchema bool
+
+	Qualified    identifier
+	Unqualified  identifier
+	MinQualified identifier
+}
+
+func newObjectName(dbType, defaultSchemaName, schemaName, tableName string) *ObjectName {
+	result := &ObjectName{
+		SchemaName:        schemaName,
+		FromDefaultSchema: schemaName == defaultSchemaName,
+		Qualified: identifier{
+			Quoted:    schemaName + "." + quote(dbType, tableName),
+			Unquoted:  schemaName + "." + tableName,
+			MinQuoted: schemaName + "." + minQuote(tableName, dbType),
+		},
+		Unqualified: identifier{
+			Quoted:    quote(dbType, tableName),
+			Unquoted:  tableName,
+			MinQuoted: minQuote(tableName, dbType),
+		},
+	}
+	result.MinQualified = lo.Ternary(result.FromDefaultSchema, result.Unqualified, result.Qualified)
+	return result
+}
+
+func (nv *ObjectName) String() string {
+	return nv.MinQualified.MinQuoted
+}
+
+func (nv *ObjectName) MatchesPattern(pattern string) (bool, error) {
+	parts := strings.Split(pattern, ".")
+	switch true {
+	case len(parts) == 2:
+		if !strings.EqualFold(parts[0], nv.SchemaName) {
+			return false, nil
+		}
+		pattern = parts[1]
+	case len(parts) == 1:
+		if !nv.FromDefaultSchema {
+			return false, nil
+		}
+		pattern = parts[0]
+	default:
+		return false, fmt.Errorf("invalid pattern: %s", pattern)
+	}
+	match1, err := filepath.Match(pattern, nv.Unqualified.Unquoted)
+	if err != nil {
+		return false, fmt.Errorf("invalid pattern: %s", pattern)
+	}
+	if match1 {
+		return true, nil
+	}
+	match2, err := filepath.Match(pattern, nv.Unqualified.Quoted)
+	if err != nil {
+		return false, fmt.Errorf("invalid pattern: %s", pattern)
+	}
+	return match2, nil
+}
+
+// <SourceTableName, TargetTableName>
+type NameTuple struct {
```

Comment:
- We should try and move this to sqlname package. That way srcdb/tgtdb can freely use this. (Will try this out when Priaynshi/I are integrating this into code)
- The "Mode" field seems a bit wrong at this layer. It should be at the lookup layer, and finally what we get is a NameTuple that knows what CurrentName is mapped to.